### PR TITLE
#1738 shopping cart component

### LINKED ISSUE #1738: Item ShoppingBasket(Cart?)Bag component
## Is your feature request related to a problem? Please describe.

Required a component to group shopping cart rows

## Describe the solution you'd like

A container component for ShoppingCartRow components

## Implementation

Probably a `FlatList`

## Describe alternatives you've considered

N/A

## Additional context

N/A

### PR DESCRIPTION
**Pull request by joshxg**
###  Branched from #1748 

Fixes #1738 

## Change summary

- Adds `PrescriptionCart` and `PrescriptionCartRow` components
- Again not great UI yet but I want to work on something else for a bit and come back a bit fresher for the UI.. I hate styling css and styling :(  🤣  
- Also a lot of callbacks are not correctly hooked up. Will be adding reducer cases and actions in a new issue without UI stuff

image

## Testing
N/A

### Related areas to think about

`PrescriptionPage` still messy. Need to sort out the `Summary`,  then can be tidied up

### CODE REVIEW

File: src/widgets/PrescriptionCart.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  },
+  titleStyles: {
+    color: SUSSOL_ORANGE,
+    fontSize: 24,
+    fontWeight: 'bold',
+    borderBottomWidth: 1,
+    borderBottomColor: SUSSOL_ORANGE,
+    marginVertical: 5,
+  },
+  flexNine: { flex: 9 },
```
Comment (mark-prins) [inline src/widgets/PrescriptionCart.js]: every time I see these in the PRs I wonder about a shared stylesheet and then dismiss the idea as too cumbersome. maybe we could chat about it though?

File: src/widgets/PrescriptionCartRow.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+  return (
+    <View>
+      <View style={localStyles.flexRow}>
+        <View style={localStyles.largeFlexRow}>
+          <StepperRow text={itemName} quantity={totalQuantity} onChangeValue={onChangeQuantity} />
+          <DetailRow details={itemDetails} />
+          <View style={{ height: 10 }} />
+          <DropdownRow
+            currentOptionText="Some direction"
```
Comment (mark-prins) [inline src/widgets/PrescriptionCartRow.js]: this is placeholder text, right?

File: src/widgets/PrescriptionCartRow.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            options={options}
+            onSelection={onOptionSelection}
+            dropdownTitle="Directions"
```
Comment (mark-prins) [inline src/widgets/PrescriptionCartRow.js]: added this PR to #1793

File: src/widgets/PrescriptionCartRow.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  onChangeQuantity: PropTypes.func.isRequired,
+  onRemoveItem: PropTypes.func.isRequired,
+  options: PropTypes.array.isRequired,
+  onOptionSelection: PropTypes.func.isRequired,
+};
+
+const localStyles = StyleSheet.create({
+  flexRow: { flex: 1, flexDirection: 'row' },
+  largeFlexRow: { flex: 10 },
+  flexOne: { flex: 1 },
```
Comment (mark-prins) [inline src/widgets/PrescriptionCartRow.js]: again, globalStyles?

Comment (joshxg) [reply src/widgets/PrescriptionCartRow.js]: you're right - but global styles or a component `<FlexView flex={x}> ... </FlexView>` ?

Comment (mark-prins) [reply src/widgets/PrescriptionCartRow.js]: oh, yeah - the `<FlexView/>` component. That's a much better idea

File: src/widgets/PrescriptionCartRow.js (hunk 2 above)
Comment (joshxg) [reply src/widgets/PrescriptionCartRow.js]: added a prop for it here: 
613200f

File: src/widgets/PrescriptionCartRow.js (hunk 4 above)
Comment (joshxg) [reply src/widgets/PrescriptionCartRow.js]: cool - will make it a new issue as I've spread this all over the place 🤣

File: src/widgets/PrescriptionCartRow.js (hunk 3 above)
Comment (joshxg) [reply src/widgets/PrescriptionCartRow.js]: thanks!

File: src/widgets/PrescriptionCart.js (hunk 1 above)
Comment (joshxg) [reply src/widgets/PrescriptionCart.js]: sure! Would love to start making things more composable to make development a lot quicker

**Pull request merged**